coderedcms/tests/test_urls.py

```diff
@@ -1,10 +1,153 @@
 import pytest
 import unittest
+
+from django.urls import reverse
 from django.test import Client
-from coderedcms.models import LayoutSettings
-from wagtail.core.models import Site
+from django.test.utils import override_settings
+
+from wagtail.core.models import Site, Page
 from wagtail.images.tests.utils import Image, get_test_image_file
 
+from coderedcms.models import LayoutSettings
+from coderedcms.tests.testapp.models import EventPage, EventIndexPage
+
+
+@pytest.mark.django_db
+class TestSiteURLs(unittest.TestCase):
+    def setUp(self):
+        self.client = Client()
+
+    @override_settings(DEBUG=False)
+    def test_404(self):
+        response = self.client.get("/testing/404/page/", follow=True)
+
+        self.assertEqual(response.status_code, 404)
+
+    def test_sitemap(self):
+        response = self.client.get("/sitemap.xml", follow=True)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response['content-type'], 'application/xml')
+
+    def test_robots(self):
+        response = self.client.get("/robots.txt", follow=True)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response['content-type'], 'text/plain')
+
+    def test_search(self):
+        response = self.client.get(reverse(
+            'codered_search'),
+            {'s': 'Test Search Query'},
+            follow=True
+        )
+
+        self.assertEqual(response.status_code, 200)
+        self.assertNotEqual(response.context['results'], None)
+
+
+@pytest.mark.django_db
+class TestEventURLs(unittest.TestCase):
+    def setUp(self):
+        self.client = Client()
+        self.root_page = Page.get_root_nodes()[0]
+
+    def test_generate_single_event(self):
+        event_page = EventPage(
+            path='/single-event/',
+            depth=1,
+            title='Single Event',
+            slug='single-event'
```

Comment:
Shouldn't the event page have a date or a recurrence associated with it, to properly test the calendar generation?